frontend/lib/identification/card_detail_view/verification_code_view.dart

```diff
@@ -57,32 +65,69 @@ class VerificationCodeViewState extends State<VerificationCodeView> {
-                clipBehavior: Clip.hardEdge,
-                shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
-                child: Stack(
-                  children: [
-                    Padding(
-                      padding: EdgeInsets.all(padding),
-                      child: qr.QrImage.withQr(
-                        qr: qrCode,
-                        version: qr.QrVersions.auto,
-                        foregroundColor: Theme.of(context).textTheme.bodyMedium?.color,
-                        gapless: false,
+            return isVerificationExpired
+                ? IconButton(
+                    icon: const Icon(Icons.refresh),
+                    color: Theme.of(context).appBarTheme.backgroundColor,
+                    onPressed: () {
+                      verifyCard(otpCode, userCode, context);
+                    },
+                  )
+                : ConstrainedBox(
+                    constraints: const BoxConstraints(maxWidth: 600, maxHeight: 600),
+                    child: Material(
+                      clipBehavior: Clip.hardEdge,
+                      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
+                      child: Stack(
+                        children: [
+                          Padding(
+                            padding: EdgeInsets.all(padding),
+                            child: qr.QrImage.withQr(
+                              qr: qrCode,
+                              version: qr.QrVersions.auto,
+                              foregroundColor: Theme.of(context).textTheme.bodyMedium?.color,
+                              gapless: false,
+                            ),
+                          ),
+                          Positioned.fill(
+                            child: AnimatedProgressbar(initialProgress: Duration(milliseconds: animationDuration)),
+                          ),
+                        ],
                       ),
                     ),
-                    Positioned.fill(
-                      child: AnimatedProgressbar(initialProgress: Duration(milliseconds: animationDuration)),
-                    ),
-                  ],
-                ),
-              ),
-            );
+                  );
           },
         );
       },
     );
   }
+
+  Future<void> verifyCard(OTPCode? otpCode, DynamicUserCode userCode, BuildContext context) async {
+    if (otpCode == null) {
+      return;
+    }
+    final projectId = Configuration.of(context).projectId;
+    final client = GraphQLProvider.of(context).value;
+    final DynamicVerificationCode qrCode = DynamicVerificationCode(
+      info: userCode.info,
+      pepper: userCode.pepper,
+      otp: otpCode.code,
+    );
+    final cardVerification = await queryDynamicServerVerification(client, projectId, qrCode);
+    if (cardVerification.valid) {
+      updateLastCardVerification(cardVerification.lastCheck);
+    } else {
+      removeInvalidCard();
```

Comment:
I think we should not remove the invalid card automatically.
In the case of an implementation/configuration error on the backend (say we accidentally always return `invalid` because we're connected to an empty DB) we don't want to automatically delete all cards in every user's app.)

Instead, we should probably persist, that the card is invalid. Probably we should persist both the timestamp and the `valid` bool.

Reply:
i updated the this in the last commit. thats why i converted it to draft yesterday. But in any case thx for the comments. 

I just have to move the verification variables from settings to the  `UserCode` Provider

Reply:
Now the card is shown as invalid if revoked or activated on other device